### PR TITLE
Llama stream multiple models supported

### PR DESCRIPTION
**Pull request by nigel-daniels**
This adds the `streamingModel` option to the streaming methods properties, this can be set to: `llama2`, `chatML`, `falcon`, or `general`. This then turns messages passed to the streaming model into the appropriate format for that model. If no option is specified it defaults to be `llama2`.

### CODE REVIEW

File: langchain/src/chat_models/llama_cpp.ts [hunk 1]
```diff
@@ -29,8 +30,14 @@ export interface LlamaCppInputs
 export interface LlamaCppCallOptions extends BaseLanguageModelCallOptions {
   /** The maximum number of tokens the response should contain. */
   maxTokens?: number;
-  /** A function called when matching the provided token array */
-  onToken?: (tokens: number[]) => void;
```
Comment (bracesproul) [inline langchain/src/chat_models/llama_cpp.ts:33]: This is not going to be backwards compatible. Instead we should mark as deprecated

File: langchain/src/chat_models/llama_cpp.ts [hunk 2]
```diff
@@ -29,8 +30,14 @@ export interface LlamaCppInputs
...
+  /** This lets the streaming method know which model we are using,
+   *  valid options are:
+   *  'llama2'	- A Llama2 model, this is the default.
+   *  'chatML'  - A ChatML model
+   *  'falcon'	- A Falcon model
+   *  'genral'  - Any other model, uses "### Human\n", "### Assistant\n" format
+   */
+  streamingModel?: string;
```
Comment (bracesproul) [inline langchain/src/chat_models/llama_cpp.ts:40]: Use a union type or a check in the constructor that verifies the input is one of these?

File: langchain/src/chat_models/tests/chatllama_cpp.int.test.ts [hunk 3]
```diff
@@ -85,14 +85,57 @@ test.skip("Test chain with memory", async () => {
...
+  const chunks = [];
+  for await (const chunk of stream) {
+    chunks.push(chunk.content);
+    console.log(chunk.content);
+  }
+
+  expect(chunks.length).toBeGreaterThan(1);
+});
+
+test.skip("test multi-mesage streaming call", async () => {
```
Comment (bracesproul) [inline langchain/src/chat_models/tests/chatllama_cpp.int.test.ts:101]: fyi if you were unaware, we don't run `int` tests in CI so no need to add `skip`

**Review by bracesproul — CHANGES_REQUESTED** [review]
Overall looks good! I have a few comments, please re-request my review once resolved!

File: langchain/src/chat_models/llama_cpp.ts (hunk 1 above)
Comment (nigel-daniels) [reply langchain/src/chat_models/llama_cpp.ts:33]: The `onToken` was included in error and was not actually applied as the eventual implementation used a raw call. I can mark it as deprecated and not used if preferred.

**Comment by nigel-daniels** [comment]
Resubmitted based off main under #3588.

**Pull request closed without merging**